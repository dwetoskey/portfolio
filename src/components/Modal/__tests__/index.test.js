// __tests__/Modal.test.js with hard coded categories
import React from 'react';
import { render, cleanup, fireEvent } from '@testing-library/react';
import '@testing-library/jest-dom/extend-expect';
import Modals from '..';

const mockToggleModal = jest.fn();
const currentPortfolio = {
  name: 'Mondegreen',
  category: 'portfolio',
  description: 'Lorem ipsum dolor sit amet, consectetur adipiscing elit. Nunc ultricie',
  index: 1
};

afterEach(cleanup);

describe('Modal component', () => {
  it('renders', () => {
    render(<Modals
      onClose={mockToggleModal}
      currentPortfolio={currentPortfolio}
    />);
  })

  it('matches snapshot DOM node structure', () => {
    const { asFragment } = render(<Modals
      onClose={mockToggleModal}
      currentPortfolio={currentPortfolio}
    />);

    expect(asFragment()).toMatchSnapshot();
  });
})

describe('Click Event', () => {
  it('calls onClose handler', () => {
    const { getByText } = render(<Modals
      onClose={mockToggleModal}
      currentPortfolio={currentPortfolio}
    />);
    fireEvent.click(getByText('Close'))

    expect(mockToggleModal).toHaveBeenCalledTimes(1);
  });
})  