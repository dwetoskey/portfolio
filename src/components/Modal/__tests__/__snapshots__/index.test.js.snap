// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Modal component matches snapshot DOM node structure 1`] = `
<DocumentFragment>
  <div />
</DocumentFragment>
`;
